cgmes/cgmes-model/src/main/java/com/powsybl/cgmes/model/CgmesOnDataSource.java

```diff
@@ -109,27 +123,59 @@ public Set<String> names() {
         try {
             // the set of names may be empty if the data source does not contain CGMES data
             Set<String> allNames = dataSource.listNames(REGEX_VALID_NAME);
-            allNames.removeIf(n -> !containsValidNamespace(n));
+            allNames.removeIf(n -> !existsInDatasource(n) || !containsValidNamespace(n));
             return allNames;
         } catch (IOException x) {
-            throw new CgmesModelException(String.format("Listing CGMES names in data source %s", dataSource), x);
+            throw new CgmesModelException(String.format(LISTING_CGMES_NAMES_IN_DATA_SOURCE, dataSource), x);
+        }
+    }
+
+    private boolean existsInDatasource(String fileName) {
+        try {
+            return dataSource.exists(fileName);
+        } catch (IOException e) {
+            return false;
         }
     }
 
     private boolean containsValidNamespace(String name) {
+        String fileExtension = name.substring(name.lastIndexOf('.') + 1);
+        if (fileExtension.equals(CompressionFormat.ZIP.getExtension())) {
+            ZipSecurityHelper.checkIfZipExtractionIsSafe(dataSource, name);
+            try (ZipInputStream is = new ZipInputStream(dataSource.newInputStream(name))) {
+                is.getNextEntry();
+                Set<String> ns = NamespaceReader.namespaces1(is);
+                return ns.contains(RDF_NAMESPACE) && ns.stream().anyMatch(CgmesNamespace::isValid);
+            } catch (XMLStreamException e) {
+                return false;
+            } catch (IOException x) {
+                throw new CgmesModelException(String.format(LISTING_CGMES_NAMES_IN_DATA_SOURCE, dataSource), x);
+            }
+        }
         try (InputStream is = dataSource.newInputStream(name)) {
             Set<String> ns = NamespaceReader.namespaces1(is);
             return ns.contains(RDF_NAMESPACE) && ns.stream().anyMatch(CgmesNamespace::isValid);
```

Comment:
You can mutualize some code by loading the `InputStream` whether it is from an xml or zip file by creating a private method doing all the work and not closing the input stream.  Keeping the try/catch outside the private method allows to then put the logic as it is. the private method can be called for the 3 times you need the `InputStream` (thank you fot the tip @olperr1 !)  

```suggestion
try (InputStream is = loadInputStream(n)) {
    return NamespaceReader.base(is);
} catch (IOException x) {
    throw new UncheckedIOException(x);
}
```
Example for the private method to load the InputStream:
```
private InputStream loadInputStream(String n) throws IOException {
        InputStream is;
        String fileExtension = n.substring(n.lastIndexOf('.') + 1);
        if (fileExtension.equals(CompressionFormat.ZIP.getExtension())) {
            ZipSecurityHelper.checkIfZipExtractionIsSafe(dataSource, n);
            is = new ZipInputStream(dataSource.newInputStream(n));
            ((ZipInputStream) is).getNextEntry();
        } else {
            is = dataSource.newInputStream(n);
        }
        return is;
    }
```

Reply:
In fact this was my first implementation (see my first commit), but then Sonar complained because the InputStream was not closed, so I had to duplicate some code...